drivers/spi/spi_nrfx_spim.c

```diff
@@ -378,17 +379,30 @@ static int transceive(const struct device *dev,
 
 	spi_context_lock(&dev_data->ctx, asynchronous, cb, userdata, spi_cfg);
 
+	(void)pm_device_runtime_get(dev);
+
 	error = configure(dev, spi_cfg);
-	if (error == 0) {
-		dev_data->busy = true;
+	if (error < 0) {
+		goto release;
+	}
 
-		spi_context_buffers_setup(&dev_data->ctx, tx_bufs, rx_bufs, 1);
-		spi_context_cs_control(&dev_data->ctx, true);
+	dev_data->busy = true;
 
-		transfer_next_chunk(dev);
+	spi_context_buffers_setup(&dev_data->ctx, tx_bufs, rx_bufs, 1);
+	spi_context_cs_control(&dev_data->ctx, true);
 
-		error = spi_context_wait_for_completion(&dev_data->ctx);
-	}
+	transfer_next_chunk(dev);
+
+	error = spi_context_wait_for_completion(&dev_data->ctx);
+
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	/* TODO: remove once nrfx automatically disables the device */
+	nrfx_spim_uninit(&dev_config->spim);
+	dev_data->initialized = false;
```

Comment:
We're moving to a model when runtime should be the only model of saving power. ie, if you want to save power make sure to set CONFIG_PM_DEVICE_RUNTIME=y. Note that, however, Nordic HAL will internally do its _magic_ anyway, at least it is planned for all peripherals.

Reply:
> We're moving to a model when runtime should be the only model of saving power

But since the non-runtime way is still available, why not keep it working?

I don't quite get it. This currently (until the nrfx driver is updated) needs to be done when the device is to be suspended, so why is this location (plus additional `#ifdef CONFIG_PM_DEVICE_RUNTIME`) better than what we had so far.